tools/interactive_marker_teleop/src/interactive_marker_teleop.cpp

```diff
@@ -16,36 +16,41 @@
  * under the License.
  */
 
-#include <ros/ros.h>
-
-#include <interactive_markers/interactive_marker_server.h>
-#include <interactive_markers/menu_handler.h>
-
-#include <tf2_ros/transform_listener.h>
-#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
-
-#include <ff_msgs/CommandStamped.h>
-#include <ff_msgs/CommandConstants.h>
-#include <ff_util/ff_names.h>
+#include <interactive_marker_teleop.h>
 
 #include <string>
 
-using visualization_msgs::InteractiveMarker;
-using visualization_msgs::InteractiveMarkerControl;
-using visualization_msgs::InteractiveMarkerFeedbackConstPtr;
-using visualization_msgs::Marker;
+InteractiveMarkerTeleop::InteractiveMarkerTeleop(ros::NodeHandle& nh) : nh(nh) {
+  // Start tf listener
+  tfListener = std::shared_ptr<tf2_ros::TransformListener>(new tf2_ros::TransformListener(tfBuffer));
 
-std::shared_ptr<interactive_markers::InteractiveMarkerServer> server;
-interactive_markers::MenuHandler menu_handler;
+  // Make the marker
+  server.reset(new interactive_markers::InteractiveMarkerServer("interactive_marker_teleop", "", false));
+  ros::Duration(0.1).sleep();
+  menu_handler.insert("Snap to Astrobee",
+                      std::bind(&InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1));
+  menu_handler.insert("Go to Position",
+                      std::bind(&InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1));
+  menu_handler.insert("Dock", std::bind(&InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1));
+  menu_handler.insert("Undock", std::bind(&InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1));
+  menu_handler.insert("Stop", std::bind(&InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1));
+
+  // menu_handler.insert("Add to plan", &InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1);  //
+  // TODO(jdekarske) would be cool to add a new marker for each station menu_handler.insert("Remove from plan",
+  // &InteractiveMarkerTeleop::processFeedback, this, std::placeholders::_1);
 
-ros::Publisher cmd_publisher;
-ros::Subscriber ack_subscriber;
+  geometry_msgs::Pose position;
+  make6DofMarker(vm::InteractiveMarkerControl::MOVE_ROTATE_3D, position);
```

Comment:
I think you could remove position from the arguments to this function and default construct the geometry_msgs::Pose in the function, although it looks like its setting the int_marker.pose, so you probably don't need it and int_marker.pose will just be default initialized.  Alternatively, you could add a default arg in the header declaration if this will be used with a non default position arg in the future